lib/arel-helpers/join_association.rb

```diff
@@ -262,6 +264,33 @@ def join_association_6_1_0(table, association, join_type, options = {})
         end
       end
 
+      def join_association_7_2_0(table, association, join_type, options = {})
+        aliases = options.fetch(:aliases, []).index_by(&:table_name)
+        associations = association.is_a?(Array) ? association : [association]
+
+        alias_tracker = ActiveRecord::Associations::AliasTracker.create(
+          table.connection_pool, table.name, {}
```

Comment:
This is the only update line - AliasTracker.create takes the connection pool now:

https://github.com/rails/rails/pull/51725/files#diff-6b3375555f1d9c06188a2676ec95c585b1e3c464c3d0bb22a0170d794575cb55

and 

https://github.com/rails/rails/pull/51725/files#diff-18a561656864ea240daf46bdb1f50faace49f9ef74b90bcf667d9bbb17fce084R954